cocos/ui/UIRichText.cpp

```diff
@@ -1492,115 +1493,151 @@ void RichText::formatText()
     }
 }
 
-static int getPrevWord(const std::string& text, int idx)
-{
-    // start from idx-1
-    for (int i=idx-1; i>=0; --i)
+namespace {
+    inline bool isUTF8CharWrappable(const StringUtils::StringUTF8::CharUTF8& ch)
     {
-		if (!std::isalnum(text[i], std::locale()))
-            return i;
+        return (!ch.isASCII() || !std::isalnum(ch._char[0], std::locale()));
     }
-    return -1;
-}
 
-static bool isWrappable(const std::string& text)
-{
-    for (size_t i = 0, size = text.length(); i < size; ++i)
+    int getPrevWordPos(const StringUtils::StringUTF8& text, int idx)
     {
-        if (!std::isalnum(text[i], std::locale()))
-            return true;
+        // start from idx-1
+        const StringUtils::StringUTF8::CharUTF8Store& str = text.getString();
+        auto it = std::find_if(str.rbegin() + (str.size() - idx + 1), str.rend(), isUTF8CharWrappable);
+        if (it == str.rend())
+            return -1;
+        return static_cast<int>(it.base() - str.begin());
     }
-    return false;
-}
 
-int RichText::findSplitPositionForWord(cocos2d::Label* label, const std::string& text)
-{
-    auto originalLeftSpaceWidth = _leftSpaceWidth + label->getContentSize().width;
+    int getNextWordPos(const StringUtils::StringUTF8& text, int idx)
+    {
+        const StringUtils::StringUTF8::CharUTF8Store& str = text.getString();
+        auto it = std::find_if(str.begin() + idx + 1, str.end(), isUTF8CharWrappable);
```

Comment:
if `idx` == str.length(), then it may have problem.

Reply:
Yes, it will be out of bounds. I will add an if condition